pyhive/hive.py

```diff
@@ -194,7 +208,12 @@ def __init__(
                 port = 10000
             if auth is None:
                 auth = 'NONE'
-            socket = thrift.transport.TSocket.TSocket(host, port)
+                
+            if 'EXTRA_USE_SSL' in configuration.keys():
+                socket = thrift.transport.TSSLSocket.TSSLSocket(host, port, cert_reqs=CERT_NONE)
```

Comment:
If Hive is configured with a restricted set of configs, you'll get this `HiveSQLException` from the server: `java.lang.IllegalArgumentException: Cannot modify EXTRA_USE_SSL at runtime. It is not in list of params that are allowed to be modified at runtime`

Easy to fix by removing the config when it's no longer needed: 

```suggestion
                socket = thrift.transport.TSSLSocket.TSSLSocket(host, port, cert_reqs=CERT_NONE)
                configuration.pop("EXTRA_USE_SSL")
```